### DISCUSSION TITLE
Can I trigger the `copy` and `paste` events?

### DISCUSSION THREAD

**Discussion by Jason5Lee** [tauri-apps/tauri#7595] [Q&A]
I'm writing an application that may read and write to the user's clipboard without the user manually doing it. So, I need to trigger these two events through the program instead of a user action. In Tauri, is this possible?

**Comment by FabianLars** [comment]
If you just want to read/write, you could use explicit clipboard apis like https://tauri.app/v1/api/js/clipboard (also available in rust, including 3rd party crates/plugins). Triggering the actual events will probably be tricky, i'd imagine the only way to actually do that is by triggering the key combination which is also a bit unreliable and bad UX imo.

**Reply by Jason5Lee** [reply]
Does it support HTML? Another option is to make `window.clipboardData` available.

**Reply by FabianLars** [reply]
> Does it support HTML? 

isn't html just text in this context?

> Another option is to make window.clipboardData available.

If it's not available already we can't really change that. We'd have to polyfill it with the tauri clipboard api (which we also do for the notification api btw).

**Reply by Jason5Lee** [reply]
Using [`getData`](https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/getData) method on the [`ClipboardEvent,clipboardData`](https://developer.mozilla.org/en-US/docs/Web/API/ClipboardEvent/clipboardData), I can set a format (which can be `text/html` so I can read formatted text as HTML). How can I read the format text as HTML if I don't have a clipboard event?

**Reply by FabianLars** [reply]
> Using [getData](https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/getData) method on the [ClipboardEvent,clipboardData](https://developer.mozilla.org/en-US/docs/Web/API/ClipboardEvent/clipboardData), I can set a format (which can be text/html so I can read formatted text as HTML)

But it's still "just" a text string right? What's the difference between reading it as text/html vs text/plain for example? I guess i digress but i've never really worked with js clipboard apis and i can't find any helpful resources about it. Just trying to understand your need to find a solution btw.

**Reply by Jason5Lee** [reply]
For example, if I copy some bold text, read as plain text would be `text`, read as html might be `<b>text<b>` .

**Reply by HuakunShen** [reply]
Yeah it's different, Electron supports all kinds of format https://www.electronjs.org/docs/latest/api/clipboard

Sometimes when we copy text, we not only need the text, but also the rich text format. 

For example, copying some web page with a black background will result in black background when pasted to word doc.

![image](https://github.com/tauri-apps/tauri/assets/33727687/491156e1-4639-49f3-b6c6-feeea48079f5)
